Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp

```diff
@@ -46,6 +46,11 @@ namespace AzFramework
         }
     }
 
+    SpawnableEntitiesManager::~SpawnableEntitiesManager()
+    {
+        AZ_Assert(m_totalTickets == 0, "Shutting down the Spawnable Entities Manager while there are still active Spawnable Tickets.");
```

Comment:
Maybe should be a warning?

Reply:
Also does this catch spawned entities after exiting game mode? (I assume it doesn't). Would it be possible to add that functionality on Spawnable API end, or should this be handled by user?

Problem we have right now, is that if something is not properly despawned after exiting Game Mode, it's impossible to then get rid off. It doesn't show up in Entity Outliner, but it can corrupt saving level if user is not paying attention. It would be nice to have some built-in mechanism to catch that.

Reply:
This should be an assert because it's a logical/programmer error and not something users can control. Warnings and errors are typically used for issues that can be fixed by addressing the data external to the executable.

Once a ticket is destroyed any entities on them are automatically destroyed as well, so the fact there's still a ticket available means there are zero or more entities present as well, with it being likely that there are entities leaking as well. I believe the Entity Context also has some checks to verify there are no more active entities on shutdown, but I might be mistaken. Note though that the above code will only trigger on shutdown, not when exiting the Game Mode as the SpawnableEntitiesManager created at boot time and destroyed during shutdown.